packages/@react-stately/list/src/useListState.ts

```diff
@@ -39,9 +39,11 @@ export interface ListState<T> {
  * of items from props, and manages multiple selection state.
  */
 export function useListState<T extends object>(props: ListProps<T>): ListState<T>  {
-  let {filter} = props;
+  let {filter, alwaysFireOnSelection, disallowEmptySelection} = props;
 
-  let selectionState = useMultipleSelectionState(props);
+  // If list disallows empty selection, fire onSelectionChange even if user selects the currently selected option. This allows
+  // the user to know if the currently selected option has been clicked in a combobox, picker, or Tab component.
+  let selectionState = useMultipleSelectionState({...props, alwaysFireOnSelection: alwaysFireOnSelection || disallowEmptySelection});
```

Comment:
Kinda iffy on the `alwaysFireOnSelection: alwaysFireOnSelection || disallowEmptySelection` bit. I added this to preserve the ActionGroup `disallowEmptySelection` behavior but I feel like that behavior is kinda weird to begin with...

Reply:
Yeah I think each component opting in would be better. I think possibly even only `useSingleSelectListState` should do that? Do we need to preserve that behavior in ActionGroup? I would consider it a bug just like in our other list components.

Reply:
Sounds good, I'll go ahead and get rid of the ActionGroup behavior then and have `useSingleSelectListState` pass `alwaysFireOnSelection: true` to `useMultipleSelectionState`.